app/code/Magento/MediaContentCatalog/Observer/Product.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentCatalog\Observer;
+
+use Magento\Catalog\Model\Product as CatalogProduct;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Model\AbstractModel;
+use Magento\MediaContentApi\Api\UpdateRelationsInterface;
+
+/**
+ * Observe the catalog_product_save_after event and run processing relation between product content and media asset
+ */
+class Product implements ObserverInterface
+{
+    private const CONTENT_TYPE = 'catalog_product';
+
+    /**
+     * @var UpdateRelationsInterface
+     */
+    private $processor;
+
+    /**
+     * @var array
+     */
+    private $fields;
+
+    /**
+     * @param UpdateRelationsInterface $processor
+     * @param array $fields
+     */
+    public function __construct(UpdateRelationsInterface $processor, array $fields)
+    {
+        $this->processor = $processor;
+        $this->fields = $fields;
+    }
+
+    /**
+     * Retrieve the saved product and pass it to the model processor to save content - asset relations
+     *
+     * @param Observer $observer
+     */
+    public function execute(Observer $observer): void
+    {
+        /** @var CatalogProduct $model */
+        $model = $observer->getEvent()->getData('product');
+        if ($model instanceof AbstractModel) {
```

Comment:
```suggestion
        if ($model instanceof \Magento\Catalog\Model\Product) {
```

Reply:
@sivaschenko will this work if this preference will be changed `    <preference for="Magento\Catalog\Api\Data\ProductInterface" type="Magento\Catalog\Model\Product" />` and `Magento\Catalog\Api\Data\ProductInterface` will have another implementation?

Reply:
No

Reply:
If not can we leave AbstractModel instance identification? I can hardly imagine the case when the `Magento\Catalog\Api\Data\ProductInterface` can have another implementation but if this possible from the di perspective it can break the process. Or I missed something?

Reply:
@naydav can you please address this question?